webpack-plugin is hard pinned to babel version

### Describe what happened

Starting with version 3.1.4, `/published/webpack-plugin/package.json` has 

```
"peerDependencies": {
        "@babel/parser": "7.24.5",
        "@babel/traverse": "7.24.5",
        "@babel/types": "7.24.5",
```

We're trying to update from 3.1.3 to 3.1.8 to resolve a security issue, but we are on a later version of babel (7.26.3). `npm 7+` sees that as an error and does not allow installing without allowing `legacy-peer-deps=true` which we shouldn't have to do.

These dependences should use something like `"@babel/parser": "^7.24.5"`

### Steps to reproduce the issue:

Use this package.json

```json
{
  "name": "datadog-issue",
  "version": "1.0.0",
  "description": "",
  "license": "ISC",
  "author": "",
  "type": "commonjs",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "devDependencies": {
    "@babel/parser": "7.26.3",
    "@datadog/webpack-plugin": "3.1.8"
  }
}
```
Do `npm i`

### Expected behaviour:

See it install correctly

### Actual behaviour:

```bash
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @datadog/webpack-plugin@3.1.8
npm error Found: @babel/parser@7.26.3
npm error node_modules/@babel/parser
npm error   dev @babel/parser@"7.26.3" from the root project
npm error
npm error Could not resolve dependency:
npm error peerOptional @babel/parser@"7.24.5" from @datadog/webpack-plugin@3.1.8
npm error node_modules/@datadog/webpack-plugin
npm error   dev @datadog/webpack-plugin@"3.1.8" from the root project
npm error
npm error Conflicting peer dependency: @babel/parser@7.24.5
npm error node_modules/@babel/parser
npm error   peerOptional @babel/parser@"7.24.5" from @datadog/webpack-plugin@3.1.8
npm error   node_modules/@datadog/webpack-plugin
npm error     dev @datadog/webpack-plugin@"3.1.8" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

### Additional context
- This was on node 24.14.0, npm 11.9.0

--- a/packages/plugins/live-debugger/package.json
+++ b/packages/plugins/live-debugger/package.json
@@ -36,9 +36,9 @@
         "typescript": "5.4.3"
     },
     "peerDependencies": {
-        "@babel/parser": "7.24.5",
-        "@babel/traverse": "7.24.5",
-        "@babel/types": "7.24.5",
+        "@babel/parser": "^7.24.5",
+        "@babel/traverse": "^7.24.5",
+        "@babel/types": "^7.24.5",
         "magic-string": "^0.30.0"
     },
     "peerDependenciesMeta": {

--- a/packages/published/esbuild-plugin/package.json
+++ b/packages/published/esbuild-plugin/package.json
@@ -85,9 +85,9 @@
         "typescript": "5.4.3"
     },
     "peerDependencies": {
-        "@babel/parser": "7.24.5",
-        "@babel/traverse": "7.24.5",
-        "@babel/types": "7.24.5",
+        "@babel/parser": "^7.24.5",
+        "@babel/traverse": "^7.24.5",
+        "@babel/types": "^7.24.5",
         "esbuild": ">=0.x",
         "magic-string": "^0.30.0"
     },

--- a/packages/published/rollup-plugin/package.json
+++ b/packages/published/rollup-plugin/package.json
@@ -88,9 +88,9 @@
         "typescript": "5.4.3"
     },
     "peerDependencies": {
-        "@babel/parser": "7.24.5",
-        "@babel/traverse": "7.24.5",
-        "@babel/types": "7.24.5",
+        "@babel/parser": "^7.24.5",
+        "@babel/traverse": "^7.24.5",
+        "@babel/types": "^7.24.5",
         "magic-string": "^0.30.0",
         "rollup": ">= 3.x < 5.x"
     },

--- a/packages/published/rspack-plugin/package.json
+++ b/packages/published/rspack-plugin/package.json
@@ -85,9 +85,9 @@
         "typescript": "5.4.3"
     },
     "peerDependencies": {
-        "@babel/parser": "7.24.5",
-        "@babel/traverse": "7.24.5",
-        "@babel/types": "7.24.5",
+        "@babel/parser": "^7.24.5",
+        "@babel/traverse": "^7.24.5",
+        "@babel/types": "^7.24.5",
         "@rspack/core": "1.x",
         "magic-string": "^0.30.0"
     },

--- a/packages/published/vite-plugin/package.json
+++ b/packages/published/vite-plugin/package.json
@@ -85,9 +85,9 @@
         "typescript": "5.4.3"
     },
     "peerDependencies": {
-        "@babel/parser": "7.24.5",
-        "@babel/traverse": "7.24.5",
-        "@babel/types": "7.24.5",
+        "@babel/parser": "^7.24.5",
+        "@babel/traverse": "^7.24.5",
+        "@babel/types": "^7.24.5",
         "magic-string": "^0.30.0",
         "vite": ">= 5.x <= 7.x"
     },

--- a/packages/published/webpack-plugin/package.json
+++ b/packages/published/webpack-plugin/package.json
@@ -85,9 +85,9 @@
         "typescript": "5.4.3"
     },
     "peerDependencies": {
-        "@babel/parser": "7.24.5",
-        "@babel/traverse": "7.24.5",
-        "@babel/types": "7.24.5",
+        "@babel/parser": "^7.24.5",
+        "@babel/traverse": "^7.24.5",
+        "@babel/types": "^7.24.5",
         "magic-string": "^0.30.0",
         "webpack": ">= 5.x < 6.x"
     },

